soci/soci_index.go

```diff
@@ -511,6 +525,9 @@ func (b *IndexBuilder) buildSociLayer(ctx context.Context, desc ocispec.Descript
 		IndexAnnotationImageLayerDigest:    desc.Digest.String(),
 	}
 	b.maybeAddDisableXattrAnnotation(&ztocDesc, toc)
+	if desc.Annotations[IndexAnnotationDisableXAttrs] == "true" {
```

Comment:
Let's just add a constant to ensure this always matches the set value in `maybeAddDisableXattrAnnotation` function.